### ISSUE TITLE
Does unstructured-pruning of tensorflow model optimization improve latency?

### ISSUE THREAD

**Issue by cha-noong** [tensorflow/model-optimization#704]
**System information**

- TensorFlow version (you are using): tf 2.3 ~2.5
- Are you willing to contribute it (Yes/No): Yes

**Motivation**

Even if I apply tfmot to my own model, the speed is the same as the existing model.

Even the sparse pre-trained model of Fast Sparse ConvNets(https://github.com/google-research/google-research/tree/master/fastconvnets) (e.g. mobilenetv2) has the same speed as the conversion to mobilenetv2 tflite in keras model zoo.

I am curious if sparsity pruning has improved latency, and if there is any improvement, I would like to know how.

Currently, tflite is being used by enabling xnnpack in versions 2.3~2.4.

**Comment by cha-noong** [comment]
The latency measurement code is as follows:

###################
from tflite_runtime.interpreter Import Interpreter
import cv2
import numpy as np
import time

interpreter = Interpreter(model_path='./mbv2_200_85_11-16b2_744.tflite')
interpreter.allocate_tensors()

input_details = interpreter.get_input_details()
output_details = interpreter.get_output_details()

img = cv2.imread("sample.jpg")
x = cv2.resize(img, dsize=(224,224))
x = x.astype("float32")
x = np.expand_dims(x, axis=0)


interpreter.set_tensor(input_details[0]['index'], x)
for i in range(1000):
    start_time = time.time()
    interpreter.invoke()
    print("time", time.time()-start_time)

####################

I ran it on Raspberry Pi 3 b+, and I checked the message that it was a delegate to xnnapack.


Also attach the used fastconvnets pre-trained model (mobilenetv2_2) and keras pre-trained model (mobilenetv2_2).

https://drive.google.com/file/d/1cC5wDEA0A9agthH9tukdhzfvBj5oNbpg/view?usp=sharing

**Comment by pedrofrodenas** [comment]
Normally the way to speed up inference speed is to prune 1x1 convolutions in network. Then this model should be converted to tflite with optimization flag. You can check that this optimization is applied checking the network in Netron and visualizing 'Densify' nodes.
